### PR TITLE
Make clippy happy.

### PR DESCRIPTION
**Pull request by jcdyer**
Here's one that does nothing but resolve the things that clippy complains about.  

Turns out clippy is a great teaching tool.

### CODE REVIEW

File: examples/hashmap.rs [hunk 1]
```diff
@@ -236,7 +235,7 @@ fn cas(args: &Args) {
 }
 
 /// A state machine that holds a hashmap.
-#[derive(Debug)]
+#[derive(Debug, Default)]
```
Comment (Hoverbear) [inline examples/hashmap.rs]: `Default` actually didn't exist when this code was written originally. :)

File: examples/hashmap.rs [hunk 2]
```diff
@@ -266,15 +265,15 @@ impl state_machine::StateMachine for HashmapStateMachine {
...
+                let old_value = &self.map.get(&key).cloned();
                 serde_json::to_string(old_value)
             },
             Put(key, value) => {
                 let old_value = &self.map.insert(key, value);
                 serde_json::to_string(old_value)
             },
             Cas(key, old_check, new) => {
-                if *self.map.get(&key).unwrap() == old_check {
+                if self.map[&key] == old_check {
```
Comment (Hoverbear) [inline examples/hashmap.rs]: I'm actually against this change. The whole `.get(&key).unwrap()` should be refactored so it doesn't crash. Using `self.map[&key]` means if the key doesn't exist we implicitly crash. At least in the old case we are more explicit about the crash.

File: src/consensus.rs [hunk 3]
```diff
@@ -399,7 +399,7 @@ impl<L, M> Consensus<L, M>
                              from,
                              leader_term);
                 self.transition_to_follower(leader_term, from, actions);
-                return self.append_entries_request(from, request, actions);
+                self.append_entries_request(from, request, actions);
```
Comment (Hoverbear) [inline src/consensus.rs]: These two changes are totally fine since `append_entries_request` returns `()`

File: src/lib.rs [hunk 4]
```diff
@@ -2,6 +2,7 @@
 #![crate_type="lib"]
 #![doc(html_logo_url = "https://raw.githubusercontent.com/Hoverbear/raft/master/raft.png")]
 #![doc(html_root_url = "https://hoverbear.github.io/raft/raft/")]
+#![cfg_attr(feature = "cargo-clippy", allow(doc_markdown))]
```
Comment (Hoverbear) [inline src/lib.rs]: Is there a reason you allowed this lint?

Comment (jcdyer) [inline src/lib.rs]: Yes.  It was requesting incorrect things, and [is known to do so](https://github.com/Manishearth/rust-clippy/wiki#doc_markdown).  Specifically, it wants anything in CapsCase to be enclosed in backticks, including names that aren't code snippets, like PostgreSQL and SQLite.

File: src/consensus.rs (hunk 3 above)
Comment (jcdyer) [inline src/consensus.rs]: Yep.  I originally changed them to exclude the semicolon, but added it back in when I saw the method didn't have a return type.

File: examples/hashmap.rs (hunk 1 above)
Comment (jcdyer) [inline examples/hashmap.rs]: Oh neat.

**Comment by Hoverbear** [comment]
Thanks again for this! :)

**Pull request merged**